src/library/dlgtagfetcher.cpp

```diff
@@ -2,6 +2,36 @@
 #include <QtDebug>
 
 #include "library/dlgtagfetcher.h"
+#include "track/tracknumbers.h"
+
+namespace {
+
+QStringList track2row(const Track& track) {
```

Comment:
I have update the dialogue only as needed. I will not implements further improvements other than passing-through of `mixxx::musicbrainz::TrackRelease` as suggested.

PS: Performance doesn't matter here. I only changed the signal types, because the status quo of using QObjects for this purpose was wrong.

Reply:
The dialogue needs to be extended anyway in the future. Whoever will do this can improve the deduplication logic and replace the QStringList. The deduplication currently depends on the displayed columns that are actually displayed and therefore must be done here.

Reply:
The simplistic deduplication logic before is really displaced and the code is difficult to follow. Now everything is co-located on the display layer were it actually matters. It's not optimal, but without a rationale for deduplication this is the best we could do for now.